### PR TITLE
Fix IndexOutOfBoundsException for CompositeByteBuf #4679

### LINKED ISSUE #4679: IndexOutOfBoundsException for CompositeByteBuf
This code worked up to netty 4.0.32, but stopped in 4.0.33.Final

        ByteBufAllocator alloc = ByteBufAllocator.DEFAULT;
        CompositeByteBuf cBuff = alloc.compositeDirectBuffer();
        cBuff.addComponent(alloc.buffer(8).writeLong(0));
        cBuff.addComponent(alloc.buffer(8).writeLong(0));
        
        // work as expected
        System.err.println("str="+ cBuff.toString(0, 16, StandardCharsets.UTF_8));
        // fails
        System.err.println("str="+ cBuff.toString(3, 13, StandardCharsets.UTF_8));

Exception:

    Exception in thread "main" java.lang.IndexOutOfBoundsException: index: 3, length: 13 (expected: range(0, 13))
	at io.netty.buffer.AbstractByteBuf.checkIndex0(AbstractByteBuf.java:1142)
	at io.netty.buffer.AbstractByteBuf.checkIndex(AbstractByteBuf.java:1137)
	at io.netty.buffer.UnpooledHeapByteBuf.internalNioBuffer(UnpooledHeapByteBuf.java:280)
	at io.netty.buffer.ByteBufUtil.decodeString(ByteBufUtil.java:558)
	at io.netty.buffer.AbstractByteBuf.toString(AbstractByteBuf.java:978)
	at Tester.main(Tester.java:24)

### PR DESCRIPTION
**Pull request by windie**
Motivation:

Modifications:

Use the correct start index

Result:

Fixes #4679

### CODE REVIEW

File: buffer/src/test/java/io/netty/buffer/ByteBufUtilTest.java [hunk 1]
```diff
@@ -187,4 +187,17 @@ private static void testDecodeString(String text, Charset charset) {
         Assert.assertEquals(text, ByteBufUtil.decodeString(buffer, 0, buffer.readableBytes(), charset));
         buffer.release();
     }
+
+    @Test
+    public void testIssue4679() {
```
Comment (Scottmitch) [inline buffer/src/test/java/io/netty/buffer/ByteBufUtilTest.java]: Can you use a more descriptive test name? something like `testToStringDoesNotThrowIndexOutOfBounds`

**Comment by Scottmitch** [comment]
@windie - Thanks for the fix. Just 1 comment.

**Comment by windie** [comment]
Updated the test name

**Comment by Scottmitch** [comment]
Cherry-picked 4.0 (6f86188) 4.1 (6fe0db4)

Thanks @windie !

**Pull request closed without merging**